app/src/org/commcare/models/FormRecordProcessor.java

```diff
@@ -108,6 +107,17 @@ public TransactionParser getParser(KXmlParser parser) {
         return updatedRecord;
     }
 
+    private void broadcastDataUpdate(AndroidTransactionParserFactory factory) {
+        Intent i = new Intent("org.commcare.dalvik.api.action.data.update");
+        i.putStringArrayListExtra("cases", factory.getCreatedAndUpdatedCases());
+        c.sendBroadcast(i, "org.commcare.dalvik.provider.cases.read");
+
+        // send explicit broadcast to CommCare Reminders App
+        i.setComponent(new ComponentName("org.commcare.dalvik.reminders",
```

Comment:
can you walk me through again briefly why the reminders application can't receive the standard broadcast instead of a custom one?

Reply:
sure, Android O+ [restricts apps to define implicit receivers in their manifest](https://developer.android.com/guide/components/broadcasts#manifest-declared-receivers). Any implicit receivers must be registered at runtime. This means that Reminders app can't listen to any implicit broadcasts without it being running in background while we want any case updates to wake Reminders app so that users don't have to repeatedly open the Reminders app manually in order for it to fetch data from CommCare.

Reply:
Right! Thank you.